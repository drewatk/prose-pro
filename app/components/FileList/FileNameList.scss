.item {
  cursor: pointer;
  background-color: var(--white-color) !important;
  color: var(--dark-color) !important;
  border: none !important;
  border-radius: 0 !important;

  overflow-x: scroll;

  &:hover,
  &:focus {
    background-color: var(--dark-white-color) !important;
    color: var(--dark-color) !important;
  }
  &:active {
    background-color: var(--dulled-primary-color) !important;
    color: var(--white-color) !important;
  }
  margin: 2px 0;
  padding: 3px 10px !important;
}

.item,
.create-form {
  border-top-right-radius: 5px !important;
  border-bottom-right-radius: 5px !important;
}
.active-item {
  background-color: var(--dulled-primary-color) !important;
  color: var(--white-color) !important;
}

.create-form {
  padding: 10px;
  background: var(--dark-white-color);
}
