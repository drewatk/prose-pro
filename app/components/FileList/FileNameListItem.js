import React from "react";
import {
  ListGroupItem,
  Dropdown,
  DropdownToggle,
  DropdownItem,
  DropdownMenu
} from "reactstrap";

class FileNameListItem extends React.Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      dropdownOpen: false
    };
  }

  toggle() {
    this.setState(prevState => ({
      dropdownOpen: !prevState.dropdownOpen
    }));
  }

  render() {
    const { file, onFileItemClick, className, onFileDeleteClick } = this.props;
    return (
      <ListGroupItem
        id="file-item"
        onClick={() => onFileItemClick(file)}
        className={className}
      >
        {file}
        <Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
          <DropdownToggle
            tag="span"
            onClick={e => {
              e.stopPropagation();
            }}
          >
            <i className="fas fa-ellipsis-h" style={{ float: "right" }} />
          </DropdownToggle>
          <DropdownMenu>
            <DropdownItem
              onClick={e => {
                e.stopPropagation();
                onFileDeleteClick(file);
              }}
            >
              Delete File
            </DropdownItem>
          </DropdownMenu>
        </Dropdown>
      </ListGroupItem>
    );
  }
}

export default FileNameListItem;
