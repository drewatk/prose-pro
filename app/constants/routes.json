{
  "HOME": "/",
  "PROJECT_SETUP": "/ProjectSetup"
}
