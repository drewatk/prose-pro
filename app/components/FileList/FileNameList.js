import React from "react";
import { connect } from "react-redux";

import { ListGroup } from "reactstrap";
import FileNameListItem from "./FileNameListItem";

import styles from "./FileNameList.scss";

export const FileNameList = props => {
  const { files, onFileItemClick, onFileDeleteClick, currentFile } = props;
  return (
    <div>
      <ListGroup>
        {files.map((file, index) => {
          return (
            <FileNameListItem
              key={index}
              onFileItemClick={onFileItemClick}
              file={file}
              className={`${styles.item} ${
                currentFile == file ? styles.activeItem : ""
              }`}
              onFileDeleteClick={onFileDeleteClick}
            />
          );
        })}
      </ListGroup>
    </div>
  );
};

const mapStateToProps = ({ currentFile }) => ({ currentFile });

export default connect(mapStateToProps)(FileNameList);
