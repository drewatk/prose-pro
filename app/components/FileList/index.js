import React from "react";
import { connect } from "react-redux";

import CreateFileForm from "../Forms/CreateFileForm";
import FileNameList from "./FileNameList";

import selectFile from "app/actions/file_selection";
import updateFiles from "app/actions/files";

const FileList = ({ files, gitAbstractions, dispatch }) => {
  return (
    <div>
      <div>
        <CreateFileForm
          onSubmit={({ fileName }) => {
            gitAbstractions
              .createFile(fileName)
              .then(() => dispatch(updateFiles(gitAbstractions.getFiles())))
              .then(() => dispatch(selectFile(gitAbstractions, fileName)))
              .catch(err =>
                console.error("Error in CreateFileForm onSubmit: ", err)
              );
          }}
        />
      </div>
      <div style={{ backgroundColor: "lightblue", height: "100%" }}>
        FileList View
      </div>
      <div>
        <FileNameList
          files={files}
          onFileItemClick={file => {
            dispatch([
              selectFile(gitAbstractions, file)
              /* load file data */
              /* load file checkpoints */
            ]);
          }}
        />
      </div>
    </div>
  );
};

const mapStateToProps = ({ files, gitAbstractions }) => ({
  files,
  gitAbstractions
});

const mapDispatchToProps = dispatch => ({ dispatch });

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FileList);
