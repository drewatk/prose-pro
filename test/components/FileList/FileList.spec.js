import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import routes from "app/constants/routes.json";
import { FileNameList } from "app/components/FileList/FileNameList";

Enzyme.configure({ adapter: new Adapter() });

describe("FileNameList", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<FileNameList />);
    expect(wrapper).toMatchSnapshot();
  });

  it("calls onFileItemClick", () => {
    const onFileItemClick = jest.fn();
    const pathname = routes.EDITOR;

    const wrapper = shallow(
      <FileNameList pathname={pathname} onFileItemClick={onFileItemClick} />
    );

    wrapper.find("#file-item").simulate("click");
    expect(onFileItemClick).toHaveBeenCalledTimes(1);
  });
});
