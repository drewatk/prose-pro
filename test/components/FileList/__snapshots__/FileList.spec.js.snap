// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileNameList matches snapshot 1`] = `
<div>
  <ListGroup
    tag="ul"
  >
    <FileNameListItem
      className="item "
      file="File1"
      key="0"
    />
    <FileNameListItem
      className="item "
      file="file2"
      key="1"
    />
    <FileNameListItem
      className="item "
      file="File3"
      key="2"
    />
  </ListGroup>
</div>
`;
